ext/http/http_next.rs

```diff
@@ -915,8 +902,7 @@ where
   let listen_properties_clone: HttpListenProperties = listen_properties.clone();
   let handle = spawn(async move {
     loop {
-      let conn = listener
-        .accept()
+      let conn = HTTP::accept_connection_from_listener(&listener)
```

Comment:
I think we could probably turn this functionality into a trait living in ext/net that any listener type can implement. I don't think we need to do it now, however.

Reply:
Yes, I wanted to refactor this to be as minimal of a change as possible. I also tried with generic connections, but it became too complicated.

Something for a follow up.